core/common/src/geometry/FrustumPlanes.ts

```diff
@@ -11,53 +11,122 @@ import { ClipPlane, Point3d, Vector3d } from "@itwin/core-geometry";
 import { Frustum } from "../Frustum";
 import { BoundingSphere } from "./BoundingSphere";
 
+const planePointIndices = [
+  [1, 3, 5],  // right
+  [0, 4, 2],  // left
+  [2, 6, 3],  // top
+  [0, 1, 4],  // bottom
+  [0, 2, 1],  // back
+  [4, 5, 6],  // front
+];
+
+function computeFrustumPlanes(frustum: Frustum): ClipPlane[] {
+  const planes = [];
+  const points = frustum.points;
+  const expandPlaneDistance = 1e-6;
+
+  for (const indices of planePointIndices) {
+    const i0 = indices[0], i1 = indices[1], i2 = indices[2];
+    const normal = Vector3d.createCrossProductToPoints(points[i2], points[i1], points[i0]);
+    normal.normalizeInPlace();
+
+    const plane = ClipPlane.createNormalAndDistance(normal, normal.dotProduct(points[i0]) - expandPlaneDistance);
+    if (!plane)
+      return [];
+
+    planes.push(plane);
+  }
+
+  assert(planes.length === 6);
+  return planes;
+}
+
 // Scratch variable used by FrustumPlanes.computeContainment.
 const planesContainingSphere = [false, false, false, false, false, false];
 
-/** Represents a frustum as 6 planes and provides containment and intersection testing
- * @internal
+/** Represents a the planes of a [[Frustum]] for testing containment and intersection.
+ * A valid frustum produces six planes. A degenerate frustum produces zero planes.
+ * @public
+ * @extensions
```

Comment:
i didnt see any changes to the extension api after this was added, which shouldve automatically occurred when you ran `rush extract-api`

Reply:
I don't have an explanation for you. If you'll recall, the most recent change to the extension api occured mysteriously in [this unrelated PR](https://github.com/iTwin/itwinjs-core/pull/4796#pullrequestreview-1217532198). The `@extensions` tags were added previously in #4737.

Reply:
In case it matters, I almost never run `rush extract-api`; I just run `npm run extract-api` in the directories of the packages I modified.
The CI job does run `rush extract-api`. Perhaps you expect it to verify no extension API changes and it's not doing so?

Reply:
Running `npm run extract-extension-api` in core/frontend produces the diffs you're expecting. That package's `extract-api` script also calls `extract-extension-api` and I can see it run when I run `extract-api` (it takes an annoyingly long time), but does not produce extension api diffs. Something is broken.

Reply:
And now CI is failing on extract-api. I'm going to revert until extension api extraction is working properly.

```
--[ FAILURE: @itwin/core-frontend ]-------------------------[ 35.38 seconds ]--

Error: src/extension/ExtensionRuntime.ts(258,3): error TS2300: Duplicate identifier 'FrustumPlanes'.
Error: src/extension/ExtensionRuntime.ts(259,3): error TS2300: Duplicate identifier 'FrustumPlanes'.
Error: src/extension/ExtensionRuntime.ts(340,3): error TS1117: An object literal cannot have multiple properties with the same name in strict mode.
Error: src/extension/ExtensionRuntime.ts(340,3): error TS2300: Duplicate identifier 'FrustumPlanes'.
```

Reply:
@aruniverse gonna worry about extension API issues separately, or have a plan to address them before this merges?

Reply:
I will handle this in a separate pr/issue.

Reply:
see https://github.com/iTwin/itwinjs-core/issues/4882